src/tools/request-merge.js

```diff
@@ -4,6 +4,51 @@ const _ = require('lodash');
 
 const requestClean = require('./request-clean');
 
+// Do a merge with case-insensitive keys, and drop empty header keys
+const caseInsensitiveMerge = (requestOne, requestTwo, requestThree) => {
+  // This creates copies/clones
+  requestOne = requestClean(requestOne);
+  requestTwo = requestClean(requestTwo);
+  requestThree = requestClean(requestThree);
+
+  // This is a very quick & efficient merge for all of request's properties
+  const mergedRequest = _.merge(requestOne, requestTwo, requestThree);
+
+  // Now to cleanup headers, we start on the last request (the one with priority) and work backwards to add the keys that don't already exist
+  // NOTE: This is done "manually" instead of a _.merge or Object.assign() because we need case-insensitivity
```

Comment:
This "why" type of note is great. Some of the other comments in this function are more "what" type comments that should be ok to nuke.